lib/irb.rb

```diff
@@ -21,36 +21,917 @@
+# but \IRB offers special treatment for values returned by assignment statements,
+# which may be:
+#
+# - Printed with truncation (to fit on a single line of output),
+#   which is the default;
+#   an ellipsis (<tt>...</tt> is suffixed, to indicate the truncation):
+#
+#     irb(main):001:0> x = 'abc' * 1000
+#     => "abcabcabcabcabcabcabcabcabcabcabcabcabcabcabcabcabcabcabcabcabcabcabcab...
+#
+# - Printed in full (regardless of the length).
+# - Suppressed (not printed at all)
+#
+# You can change the initial behavior by:
+#
+# - Adding to the configuration file: <tt>IRB.conf[:ECHO_ON_ASSIGNMENT] = false</tt>.
+#   (The default value for this entry is +niL+, which means the same as +:truncate+.)
+# - Giving command-line option <tt>--noecho-on-assignment</tt>
+#   or <tt>--echo-on-assignment</tt>.
+#   (The default is <tt>--truncate-echo-on-assigment</tt>.)
+#
+# See {Notes on Initialization Precedence}[rdoc-ref:IRB@Notes+on+Initialization+Precedence].
+#
+# During the session, you can change the current setting
+# with configuration method <tt>conf.echo_on_assignment=</tt>
+# (set to +true+, +false+, or +:truncate+).
+#
+# By default, \IRB formats returned values by calling method +inspect+.
+#
+# You can change the initial behavior by:
+#
+# - Adding to the configuration file: <tt>IRB.conf[:INSPECT_MODE] = false</tt>.
+#   (The default value for this entry is +true+.)
+# - Giving command-line option <tt>--noinspect</tt>.
+#   (The default is <tt>--inspect</tt>.)
+#
+# See {Notes on Initialization Precedence}[rdoc-ref:IRB@Notes+on+Initialization+Precedence].
+#
+# During the session, you can change the setting using method <tt>conf.inspect_mode=</tt>.
+#
+# === Multiline Output
+#
+# By default, \IRB prefixes a newline to a multiline response.
+#
+# You can change the initial default value by adding to the configuation file:
+#
+#   IRB.conf[:NEWLINE_BEFORE_MULTILINE_OUTPUT] = false
+#
+# During a session, you can retrieve or set the value using
+# methods <tt>conf.newline_before_multiline_output?</tt>
+# and <tt>conf.newline_before_multiline_output=</tt>.
+#
+# Examples:
+#
+#   rb(main):001:0> conf.inspect_mode = false
+#   => false
+#   irb(main):002:0> "foo\nbar"
+#   =>
+#   foo
+#   bar
+#   irb(main):003:0> conf.newline_before_multiline_output = false
+#   => false
+#   irb(main):004:0> "foo\nbar"
+#   => foo
+#   bar
+#
+# === Evaluation History
+#
+# By default, \IRB saves no history of evaluations (returned values),
+# and the related methods <tt>conf.eval_history</tt>, <tt>_</tt>,
+# and <tt>__</tt> are undefined.
+#
+# You can turn on that history, and set the maximum number of evaluations to be stored:
+#
+# - In the configuration file: add <tt>IRB.conf[:EVAL_HISTORY] = _n_</tt>.
+#   (Examples below assume that we've added <tt>IRB.conf[:EVAL_HISTORY] = 5</tt>.)
+# - In the session (at any time): <tt>conf.eval_history = _n_</tt>.
+#   [BUG] Raises exception "undefined method `push' for nil:NilClass (NoMethodError)."
```

Comment:
This doesn't bug out for me (IRB 1.8.3 on macOS):

```
irb(main):003> conf.eval_history = 10
=> 10
irb(main):004> "hi"
=> "hi"
irb(main):005> _
=> "hi"
irb(main):006> __
=> 
3 10
4 "hi"
5 "hi"
```

Reply:
Removed.  (Don't what I did to make me think this.)